### PR TITLE
executable to just setup and run the demo

### PR DESCRIPTION
**Pull request by dollschasingmen**

### CODE REVIEW

**Comment by dollschasingmen** [comment]
@rfroetscher

File: bin/executables/lib/playground.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  class Playground
+    def self.setup(host, db, port, user, password, options = {})
+      config_path = options[:config_path] || '/tmp/aleph/demo/configuration'
+
+      seed_db = options[:seed_db]
+
+      config_generator = ConfigGenerator.new(config_path, 'playground')
+
+      # redis envs
+      config_generator.merge_envs(redis_url: 'redis://localhost:6379')
```
Comment (rfroetscher) [inline bin/executables/lib/playground.rb]: why isnt this stuff in the default_yml_config below?

Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: i think mostly because its an env var and not a config per se ...

Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: hmm that default_yml_config method doesn't really make that much sense anymore i might just expose that logic into `playground.rb`

Comment (rfroetscher) [inline bin/executables/lib/playground.rb]: does it make sense to have default_encv_configs somewhere?

File: bin/aleph [hunk 2]
```diff
@@ -33,6 +33,26 @@ class CommandParser
         @options[:worker_processes]  = w
       end
 
+      opts.on('-H', '--redshift-host REDSHIFT_HOST', 'Redshift Hostname') do |rhost|
```
Comment (rfroetscher) [inline bin/aleph]: is there a reason you chose for these to be options and not env variables?

File: bin/executables/lib/playground.rb (hunk 1 above)
Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: what do you mean exactly?

Comment (rfroetscher) [inline bin/executables/lib/playground.rb]: your approach sounds fine. just makes sense for all the defaulting to be happening in one place.

Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: yea ... i think its all happening here though. at least w/r/t to a demo/minimal setup context

File: bin/aleph (hunk 2 above)
Comment (dollschasingmen) [inline bin/aleph]: easier? seems more confusing to have to put the all host info in one way and specify the secrets another way.

File: bin/executables/lib/playground.rb (hunk 1 above)
Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: i mean its all happening in playground.rb now, aka the actual values are specified there. and the config_generator is just a tool which writes the specified configs and and envs

Comment (rfroetscher) [inline bin/executables/lib/playground.rb]: what about that `properties` part?

Comment (dollschasingmen) [inline bin/executables/lib/playground.rb]: i moved it into playground: https://github.com/lumoslabs/aleph/pull/8/commits/895e01fb20205be51b9905463002637cc60e9ce1

**Comment by rfroetscher** [comment]
in general this seems fine, however i am beginning to wonder about our breakdown of config files versus arguments versus env variables. 

it feels like if we think about this holistically, we should just allow _anything_ to be config'd with env variables or a written config file. generating a default config file should be easy, and overriding defaults with env varibales (for secrets and for demo purposes) or by providing a alternate written config (maybe more for prod) would all use the same system.

thoughts?

File: docker_playground/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# log location
+RUN mkdir -p /var/log/aleph
+ENV SERVER_LOG_ROOT /var/log/aleph
+
+# make temp writeable
+RUN chmod 777 /tmp
+
+RUN gem install aleph_analytics
+EXPOSE 3000
```
Comment (dollschasingmen) [inline docker_playground/Dockerfile]: @rfroetscher so this dockerfile sets up the env and installs the gem. and then to use it `docker run -ti -p 3000:3000 lumos/aleph-playground /bin/bash -c "aleph setup_minimal -H {host} -D {db} -p {port} -U {user} -P {password}; redis-server & aleph run_demo"
`

Comment (rfroetscher) [inline docker_playground/Dockerfile]: cooooooool

**Pull request merged**